image/image.go

```diff
@@ -26,6 +26,9 @@ import (
 // validates the manifest pointed to by the given ref
 // or returns an error if the validation failed.
 func ValidateLayout(src, ref string) error {
+	if _, err := os.Stat(src); err != nil {
```

Comment:
My thought is that we should check function's legal or illegal argument in its self. It seems to be inappropriate to throw the checking out to other functions. If other functions work out to remove the checking, it is hard to be aware, and the bug will be recurrent.

> it's possible, even if unlikely, that that path is removed after this check but before being opened
Right, so we check it in `(w *pathWalker) walk()` which can protect the execution from the the case.

If my comment fully cover your idea? Of course hopefully we can present a nice solution. Thanks so much.

/cc @stevvooe @runcom @s-urbaniak